ykrt/src/mt.rs

```diff
@@ -139,9 +169,78 @@ impl MT {
         }
     }
 
-    /// `Location`s represent a statemachine: this function transitions to the next state (which
-    /// may be the same as the previous state!). If this results in a compiled trace, it returns
-    /// `Some(pointer_to_trace_function)`.
+    pub fn set_loc_code_ptr(loc: *mut Location, code: *const c_void) {
+        let mt = MT::global();
+        let js = &mut mt.inner.jit_state.lock().unwrap();
+        js.trace_code = code as usize;
+    }
+
+    /// FIXME: This is an interim version of `transition_location()` with (over-)simplified logic.
+    /// The state machines of `Location`s isn't properly hooked in, but this is the proposed
+    /// interface.
+    ///
+    /// This function is called once upon each execution of the control point. The internal state
+    /// of the supplied `Location` (if any) may be mutated, maybe moving the `Location` to a
```

Comment:
this doesn't really parse after `maybe moving`. Is it: "for example by moving the Location to a different state than its state machine"?

Reply:
@vext01 Has this been addressed?

Reply:
This one is addressed in here: https://github.com/ykjit/yk/pull/447/commits/c2415f256c08431217aed5f7effc83cfc69d10eb#diff-9fa85f3ab0ab08c2bb4337dd17d47605f520c2780158e50e66eff7cd9dc1b29bR183